spec/snippets-spec.js

```diff
@@ -0,0 +1,1148 @@
+            editor.insertText("foo");
+            expect(editor.lineTextForBufferRow(0)).toBe("with placeholder tesfoo");
+
+            simulateTabKeyEvent();
+            expect(editor.lineTextForBufferRow(1)).toBe("without placeholder var quicksort = function () {");
+            editor.insertText("test");
+            expect(editor.lineTextForBufferRow(1)).toBe("without placeholder testvar quicksort = function () {");
+            editor.backspace();
+            editor.insertText("foo");
+            expect(editor.lineTextForBufferRow(1)).toBe("without placeholder tesfoovar quicksort = function () {");
+          });
+        });
+      });
+
+      describe("when the snippet contains hard tabs", () => {
+        describe("when the edit session is in soft-tabs mode", () => {
+          it("translates hard tabs in the snippet to the appropriate number of spaces", () => {
+            expect(editor.getSoftTabs()).toBeTruthy();
+            editor.insertText("t3");
+            simulateTabKeyEvent();
+            expect(editor.lineTextForBufferRow(1)).toBe("  line 2");
+            expect(editor.getCursorBufferPosition()).toEqual([1, 8]);
+          });
+        });
+
+        describe("when the edit session is in hard-tabs mode", () => {
+          it("inserts hard tabs in the snippet directly", () => {
+            editor.setSoftTabs(false);
+            editor.insertText("t3");
+            simulateTabKeyEvent();
+            expect(editor.lineTextForBufferRow(1)).toBe("\tline 2");
+            expect(editor.getCursorBufferPosition()).toEqual([1, 7]);
+          });
+        });
+      });
+
+      describe("when the snippet prefix is indented", () => {
+        describe("when the snippet spans a single line", () => {
+          it("does not indent the next line", () => {
+            editor.setCursorScreenPosition([2, Infinity]);
+            editor.insertText(' t1');
+            atom.commands.dispatch(editorElement, 'snippets:expand');
+            expect(editor.lineTextForBufferRow(3)).toBe("    var pivot = items.shift(), current, left = [], right = [];");
+          });
+        });
+
+        describe("when the snippet spans multiple lines", () => {
+          it("indents the subsequent lines of the snippet to be even with the start of the first line", () => {
+            expect(editor.getSoftTabs()).toBeTruthy();
+            editor.setCursorScreenPosition([2, Infinity]);
+            editor.insertText(' t3');
+            atom.commands.dispatch(editorElement, 'snippets:expand');
+            expect(editor.lineTextForBufferRow(2)).toBe("    if (items.length <= 1) return items; line 1");
+            expect(editor.lineTextForBufferRow(3)).toBe("      line 2");
+            expect(editor.getCursorBufferPosition()).toEqual([3, 12]);
+          });
+        });
+      });
+
+      describe("when the snippet spans multiple lines", () => {
+        beforeEach(() => {
+          editor.update({autoIndent: true});
```

Comment:
This used to return, which caused a timeout on my machine. I think I saw @rafeca encountered this before at some point

Reply:
This was fixed in upstream changes (#290) that were merged in after running the conversion